public/app/plugins/panel/table/TablePanel.tsx

```diff
@@ -26,6 +44,133 @@ export function TablePanel(props: Props) {
 
   let tableHeight = height;
 
+  const threshold = useMemo(() => {
+    const timeField = main.fields.find((f) => f.type === FieldType.time);
+
+    if (!timeField) {
+      return 0;
+    }
+
+    return calculateAroundPointThreshold(timeField);
+  }, [main]);
+
+  const onDataHoverEvent = useCallback(
+    (evt: DataHoverEvent) => {
+      if (evt.payload.point?.time && evt.payload.rowIndex !== undefined) {
+        const timeField = main.fields.find((f) => f.type === FieldType.time);
+        const time = timeField!.values[evt.payload.rowIndex];
+
+        // If the time value of the hovered point is around the time value of the
+        // row with same index, highlight the row
+        if (isPointTimeValAroundTableTimeVal(evt.payload.point.time, time, threshold)) {
+          setRowHighlightIndex(evt.payload.rowIndex);
+          return;
+        }
+
+        // If the time value of the hovered point is not around the time value of the
+        // row with same index, try to find a row with same time value
+        const matchedRowIndex = timeField!.values.findIndex((t) =>
+          isPointTimeValAroundTableTimeVal(evt.payload.point.time, t, threshold)
+        );
+
+        if (matchedRowIndex !== -1) {
+          setRowHighlightIndex(matchedRowIndex);
+          return;
+        }
+
+        setRowHighlightIndex(undefined);
+      }
+    },
+    [main.fields, threshold]
+  );
+
+  useEffect(() => {
+    if (!config.featureToggles.tableSharedCrosshair) {
+      return;
+    }
+
+    if (
+      !panelContext.sync ||
+      panelContext.sync() === DashboardCursorSync.Off ||
+      !hasTimeField(main) ||
+      options.footer?.enablePagination
+    ) {
+      return;
+    }
+
+    const subs = new Subscription();
+
+    subs.add(
+      panelContext.eventBus
+        .getStream(DataHoverEvent)
+        .pipe(throttleTime(50))
```

Comment:
I would suggest replacing `throttleTime` with `debounceTime` and increasing the timeout to something like 250-500 ms.

throttleTime will take one event and ignore the rest for 50 milliseconds. debounceTime will ignore events until there hasn't been a new event in 50 milliseconds.

throttleTime has the effect that if a user drags his/her mouse over a bunch of points and then stops, it might not be the point the mouse is over when the users stops dragging.

debounceTime has the effect that if the user drags over a bunch of points they will be ignored until the mouse is stopped.

Reply:
Great idea! Will look into that, thank you!

Reply:
I've tested out the `debounceTime` function but it adds a weird delay into the mix. My understanding of this func is that it only emits a value after a given time period passes without any new emits. So hovering over a point in a graph will result in a row highlight only after the timeout passes with no new events(e.g.: after half a second for `debounceTime(500)`).
 
I would leave the throttle as it is to be in sync with how `GraphNG` deals with `DataHoverEvent`.

Reply:
I think the delay is something we would want here, the constant redrawing of the table is quite resource intense. Maybe lower the debounceTime from 500 to 100-200?

Reply:
That makes sense, i agree. Switched to a debounceTime of 100.

Reply:
@torkelo wants to switch the default to setting of dashboards to shared cursor, so performance here will be important.

https://github.com/grafana/grafana/pull/78636

Reply:
i think we already throttle these eventbus events during emission. i think adding a second level of thottling wont be too good. maybe we can just increase the throttle time during emission?